src/mrinufft/operators/autodiff.py

```diff
@@ -152,3 +152,94 @@ def samples(self, value):
     def __getattr__(self, name):
         """Forward all other attributes to the nufft_op."""
         return getattr(self.nufft_op, name)
+
+
+class BatchedNufftAutoGrad(MRINufftAutoGrad):
+    """
+    A batched wrapper for NUFFT operator with support for autodifferentiation
+    and varying sensitivity maps (smaps) across batches.
+
+    Parameters
+    ----------
+    nufft : object
+        An instance of a standard NUFFT operator.
+    batch_size : int
+        Number of batches to process simultaneously.
+    **kwargs : dict, optional
+        Additional arguments.
+
+    Notes
+    -----
+    #TODO Future improvements may include support for varying trajectories across batches.
```

Comment:
Don't put this in the Docstring (A comment is fine though)

Inside the note you can details the math if you wish, and provide an example of usage as well